### ISSUE TITLE
contributing

### ISSUE THREAD

**Issue by eynan** [Trisfald/weasel#17]
Hello @Trisfald . 
I'm renan and I'm learning Rust, I saw your project and would like to contribute for him. But i never contributed for opensource project and i don't know how to begin. Your https://github.com/Trisfald/weasel/blob/master/CONTRIBUTING.md is more generic and I don't find this project board with tasks. 
How to would start help you?

**Comment by Trisfald** [comment]
Hello Renan!
First of all thanks for your interest in the project.
You are right, there were no tasks planned for the moment - just a rough idea expressed in the roadmap. I created a new board (not an expert on that) and tried to put some tasks over there.

I'll briefly summarize the my ideas here.

- There are a few design tasks. Not much coding here, but still a worthwhile effort:
  - How to do Statistics/abilities for teams
  - Find use cases that are not covered 
  - How to do a graveyard for entities
- Some benchmarks (like events per second)
- More examples
  - Client / server over network
  - Reactive abilities
- Add more system metrics
- Fix a small issue with the `Originated` decorator.

Plus, I'm currently working on long lasting status effects.

I hope this helps!

**Comment by rnbguy** [comment]
Sorry for replying to a closed issue. Covid-19 made me be interested in card games. So I was looking for ways to implement some card/board games to play with my friends online. Rust is my favorite language right now, so I am so excited to find `weasel`.

But the example implementations look a bit complicated. Can you please provide with? 
1. 1-2 example(s) of simple card games.
2. A written/figure model of `weasel`'s work-flow.

You don't have to be detailed. I know a user guide is on your roadmap. I am willing to contribute once I understand your codebase.

**Comment by Trisfald** [comment]
Hello @rnbguy ,
Weasel is designed for typical turn-battle games, but it can support card games nonetheless. On a side note, some type names may sound 'awkward' in the context of card games (creature, for instance).

I took this chance to write a new example. It shows a very simple game and network play for three players. 
I kept the code for the game logic simple as I could, yet I admit it didn't turn out as straightforward as I wished. The big complexity issue is handling the game flow; this's especially true for multiplayer games where the transfer medium is asynchronous. 
Hopefully it will help understand how to make a card game at least a little. Keep in mind that the example's network code is very bare-bone. Error handling is not done in a proper way - it would have bloated the source code.

The example is here: https://github.com/Trisfald/weasel/tree/master/examples/king_of_the_hill.
In addition, I started to write down some work flow diagrams: https://github.com/Trisfald/weasel/wiki/Work-flows. I'll add more in the future.

I hope the resources above will help you to better understand and use weasel. If you have further questions feel free to ask!
